app/mm-redux/reducers/entities/apps.ts

```diff
@@ -14,11 +14,102 @@ export function bindings(state: AppBinding[] = [], action: GenericAction): AppBi
         validateBindings(action.data);
         return action.data || [];
     }
+    case AppsTypes.CLEAR_APP_BINDINGS:
+        if (state.length > 0) {
+            return [];
+        }
+        return state;
+    default:
+        return state;
+    }
+}
+
+export function bindingsForms(state: AppCommandFormMap = {}, action: GenericAction): AppCommandFormMap {
+    switch (action.type) {
+    case AppsTypes.RECEIVED_APP_BINDINGS:
+        if (Object.keys(state).length) {
+            return {};
+        }
+        return state;
```

Comment:
Can you explain the two different situations here?

Reply:
If we were to return always a {}, if the state is already empty, we would trigger a state change (which is not desirable), since it would be a new empty object.

The high level reading of this could be "if the state is not clear, clear it".